integreat_cms/cms/migrations/0033_event_datetime.py

```diff
@@ -0,0 +1,76 @@
+# Generated by Django 3.2.14 on 2022-08-21 13:03
+
+import datetime
+from django.db import migrations, models
+
+# pylint: disable=unused-argument
+def start_and_end_init(apps, schema_editor):
+    """
+    Initalize the new event datetime fields 'start' and 'end' from the respective
+    existing fields *_date and *_time.
+
+    :param apps: The configuration of installed applications
+    :type apps: ~django.apps.registry.Apps
+
+    :param schema_editor: The database abstraction layer that creates actual SQL code
+    :type schema_editor: ~django.db.backends.base.schema.BaseDatabaseSchemaEditor
+    """
+    events = apps.get_model("cms", "Event")
+    for event in events.objects.all():
+        event.start = datetime.datetime.combine(event.start_date, event.start_time)
+        event.end = datetime.datetime.combine(event.end_date, event.end_time)
+
+
+class Migration(migrations.Migration):
+    """
+    Update the datetime fields from the event model
+    """
+
+    dependencies = [
+        ("cms", "0032_region_bounding_box"),
+    ]
+
+    operations = [
+        migrations.AlterModelOptions(
+            name="event",
+            options={
+                "default_permissions": ("change", "delete", "view"),
+                "default_related_name": "events",
+                "ordering": ["start"],
+                "permissions": (("publish_event", "Can publish events"),),
+                "verbose_name": "event",
+                "verbose_name_plural": "events",
+            },
+        ),
+        migrations.AddField(
+            model_name="event",
+            name="end",
+            field=models.DateTimeField(
+                default=datetime.datetime.now, verbose_name="end"
+            ),
+        ),
+        migrations.AddField(
+            model_name="event",
+            name="start",
+            field=models.DateTimeField(
+                default=datetime.datetime.now, verbose_name="start"
+            ),
+        ),
+        migrations.RunPython(start_and_end_init),
```

Comment:
@timoludwig  
So this has no effect in the normal migration? Since the fields exist when we call the script? But are they not deleted right after in any case? Maybe I´m wrong here. But from the logic standpoint I see it as: If the fields start_time etc. do not exist, create them and then delete them right after?

Reply:
I'm not sure whether I understand your comment correctly... Yes, the reversing code has no effect on the normal migration, but only because it is not executed, not because it has no side effects...
The first arguments of [RunPython()](https://docs.djangoproject.com/en/3.2/ref/migration-operations/#runpython) is called only when the migration is applied, and the second argument only when the migration is unapplied. When [reversing migrations](https://docs.djangoproject.com/en/3.2/topics/migrations/#reversing-migrations), the opposite operations are executed in the reverse order, so:
#### Applying this migration:
1. Create new `start` and `end` fields
2. Execute `start_and_end_init()` to populate new fields with the combined values of the old fields
3. Make `start` and `end` fields non-nullable
4. Remove old `start_date`, `start_time`, `end_date` and `end_time` fields
#### Reversing this migration:
1. Create old `start_date`, `start_time`, `end_date` and `end_time` fields
2. Make `start` and `end` fields nullable
3. Execute `start_and_end_reverse()` to populate the old fields with the split values of the new fields
4. Remove new `start` and `end` fields

Reply:
Ah got it! I miss understand that the second argument of RunPython is only run when the migration is unapplied. That is actually a cool concept, I have to dive deeper into the django migrations at some point I guess :D

Reply:
Added it!